src/Components/Login.js

```diff
@@ -1,16 +1,123 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import styled from 'styled-components';
+import Button from './Button';
+import colors from "../constants/colors";
+import { Link, useNavigate } from "react-router-dom";
+import { useQuery, gql, useLazyQuery } from "@apollo/client";
+
+
+const GET_USER = gql`
+  query getAUser($email: String!) {
+    details: getAUser(email: $email) {
+      id
+      email
+      name
+			items {
+				id
+				name
+				description
+				available
+				category
+				status
+				amount
+			}
+    }
+  }
+`;
+
```

Comment:
moving this query and restructuring it seems a lot more streamlined- great choice!